subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.java

```diff
@@ -144,9 +158,47 @@ public CheckstyleReports reports(Action<? super CheckstyleReports> configureActi
         return reports;
     }
 
+    /**
+     * JavaLauncher for toolchain support
+     * @since 7.5
+     */
+    @Incubating
+    @Nested
+    public Property<JavaLauncher> getJavaLauncher() {
+        return javaLauncher;
+    }
+
     @TaskAction
     public void run() {
-        CheckstyleInvoker.invoke(this);
+        runWithProcessIsolation();
+    }
+
+    private void runWithProcessIsolation() {
```

Comment:
So the checkstyle task now always forks a worked. Did you consider making this configurable, so you may choose to use project isolation or only classloader isolation?

Reply:
I checked the performance of `classloader isolation`. And it has worse performance. Execution time went from 2 minutes to 3 minutes. Also, the performance test failed since a different daemon was started for some reason.
https://builds.gradle.org/buildConfiguration/Gradle_Master_Util_Performance_AdHocPerformanceScenarioLinux/49847809?buildTab=tests&expandedTest=build%3A%28id%3A49847809%29%2Cid%3A6645.

Maybe a reason for that is that we don't cache classloaders with `classloader isolation` and there is more memory pressure.